[changelog] Add bootstrap listener implementation to after image changelog consumer

sixpluszero opened the pull request:
<!--
Add a list of affected components in the PR title in the following format:
[component1]...[componentN] Concise commit message

Valid component tags are: [da-vinci] (or [dvc]), [server], [controller], [router], [samza],
[vpj], [fast-client] (or [fc]), [thin-client] (or [tc]), [changelog] (or [cc]),
[pulsar-sink], [producer], [admin-tool], [test], [build], [doc], [script], [compat]

Example title: [server][da-vinci] Use dedicated thread to persist data to storage engine

Note: PRs with titles not following the format will not be merged
-->

## [changelog] Add bootstrap listener implementation to after image changelog consumer
This PR adds bootstrap listener on regular changelog consumer and as well as specific record support.
User can override the listener bootstrap completed notification behavior to perform any action

## How was this PR tested?
<!--
If you're unsure about what to test, where to add tests, or how to run tests, please feel free to ask. We'd be happy to help.
-->
Added an integration test to verify the listener is working.
## Does this PR introduce any user-facing changes?
<!--
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If no, choose 'No'.
-->
- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

FelixGV reviewed (commented):
Thanks Jialin. I left several minor comments. Can you also check the unit test coverage please...?

sixpluszero commented:
Hi @FelixGV  thanks for the review! As discussed with our customer, we think this listener implementation is too complicated for single user to use and is an overkill. Instead a simple change in #1016 should just work for them. I believe that new PR is fairly easily and has addressed some of the comment you posted - list Java doc. If possible could you take a look at that PR?
I plan to close the current PR now.
